packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,207 @@
+  input => input.reduce((total, curr) => total + curr, 0),
+);
+
+// Schedule like above.
+```
+
+You can also return a Promise or an Observable, as jobs are asynchronous. This helper will set 
+progress to 0, start and end events appropriately, and will pass in a logger. It will also manage
+channels automatically (see below).
+
+A more complex job can be declared like this:
+
+```typescript
+import { Observable } from 'rxjs';
+import { JobRegistry } from './api'
+
+// Show progress with each count. Output "more" in a channel.
+export const count = jobs.createJob<number, number>(
+  // Receive a context that contains additional methods, like progress, logger and channels.
+  (input: number, { progress, channels }) => new Observable<number>(o => {
+    let i = 0;
+    function doCount() {
+      o.next(i++);
+      progress(i / input);
+      channels['side'].next('more');
+  
+      if (i < input) {
+        setTimeout(doCount, 100);
+      } else {
+        o.complete();
+      }
+    }
+    setTimeout(doCount, 100);
+  }),
+  {
+      input: { type: 'number' },
+      output: { type: 'number' },
+      channels: {
+        'side': { type: 'string', const: 'more' },
+      },
+  },
+);
+
+declare const registry: JobRegistry;
+// Register this job with `registry.register('count', count)`.
+
+const job = registry.schedule('count');
+job.channels['side'] && job.channels['side'].subscribe((x: json.JsonValue) => console.log(x));
+const c = job.getChannel<string>('side', { type: 'string' });
+if (c) {
+  c.subscribe((x: string) => console.log(x));
+}
+```
+
+## Job Extension
+A Job can extend other jobs, meaning their inputs and outputs have to be a superset of the job 
+extended. This is useful for making compatible jobs that can be replaced by others.
+
+This is supported by the SimpleJobRegistry, but other registries would support it as well.
+
+## Dispatchers
+Dispatchers are an additional way to synchronize and execute logic on scheduling the job, from the 
+side of the scheduler. This logic will always apply in the thread the job is scheduled, and not
+where the job is created/registered.
+
+Job Dispatchers are a job helper that redirect to different jobs. To create a dispatcher, use the
+`createDispatcher()` function:
+
+```typescript
+import { jobs } from '@angular-devkit/core';
+
+// A dispatcher that installs node modules.
+const dispatcher = jobs.createDispatcher({
+  name: 'node-install',
+  input: {
+    properties: {
+      moduleName: { type: 'string' },
+    },
+  },
+  output: { type: 'boolean' },
+});
+
+// Extending makes sure the input and output matches the dispatcher.
+const npmInstall = jobs.createJob(/* ... */, { name: 'npm-install', extends: dispatcher });
+const yarnInstall = jobs.createJob(/* ... */, { name: 'yarn-install', extends: dispatcher });
+const pnpmInstall = jobs.createJob(/* ... */, { name: 'pnpm-install', extends: dispatcher });
+
+declare const registry: jobs.SimpleJobRegistry;
```

Comment:
`declare` here is not correct, since you are not writing a declaration file.

Reply:
This is an example in the README. It's meant to provide context on the type, not be copy-pasted as is.